commit-message.md

```diff
@@ -0,0 +1,36 @@
+# Guidelines to create great commit messages
+
+
+We use [GitCop](https://gitcop.com) to check that commit messages are
+properly written. The rules are the following:
+
+* The first line of a commit message, called the subject line should
+  not be more than 80 characters long.
+
+* The commit message should end with the following trailers:
+
+  ```
+  Licence: MIT
+  Signed-off-by: User Name <email@address>
+  ```
+
+  where "User Name" is the author's real name and email@address one of
```

Comment:
On Mon, Jun 08, 2015 at 01:28:01AM -0700, rht wrote:
> @wking Contribution signing (any of commit/tag/pr) is much preferred
> even in the non-pseudonymous case. It just happens that it is still
> not practical nowadays (for large scale projects).

I don't understand why it's not practical.  I think Gerwitz explains
things well in [1], but the main issues he points out include:

a. What good are signatures by a one-off key?
b. How do you handle signatures in the context of rebases or
   additional commit-message trailers?
c. How do you aggregate signatures as commits trickle up a lieutenant
   tree?  Or equivalently, as external auditors review the changes.

and I don't see those being addressed in the near future.  In any
case, those features would be nice, but don't seem to be major
adoption blockers.  I expect the major issue with adoption is just
lack of motivation, just like it is for the majority of email users
not using OpenPGP.

> When it becomes practical it will enable pseudonymous contributor to
> verify itself.

For this purpose, I think it's practical now.
 
> From your viewpoint, I wonder if this is an unnecessary side effect,
> or should be prevented by any means? For the latter, e.g. every key
> has to be tied to a gov id.

I haven't looked up any previous case history (and I don't even know
if previous cases exist), but I expect tying the singing key to a
government ID isn't neccessary.  For proving ownership of a commit,
I'd expect a court to accept either:

a. The commit is written by Trevor King, and here's my passport
   proving that that's me, and here's a bit of background
   differentiating me from other Trevor Kings who may be claiming that
   commit.

b. The commit is written by foobar, and here's the signature by
   foobar's key, and here's the fooboar-signed copy of that challenge
   text you gave me yesterday, proving that I have access to foobar's
   secret key and am therefore likely foobar.

I personally prefer (a), but I think (b) is a valid choice for folks
who prefer a pseudonym and expect to only address courts that
recognize pseudonyms.

[1]: http://mikegerwitz.com/papers/git-horror-story.html